iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergOutputCommitter.java

```diff
@@ -77,7 +77,7 @@
   );
 
   private static final PartitionSpec PARTITIONED_SPEC =
-      PartitionSpec.builderFor(CUSTOMER_SCHEMA).bucket("customer_id", 3).build();
+      PartitionSpec.builderFor(CUSTOMER_SCHEMA).identity("customer_id").build();
```

Comment:
I remember that you have already told me why is this change needed. Could you please refresh my memory?

Reply:
Most of the write path is mocked out, Hive queries are not committed to test writes in this test class. That means no optimizers can run either, and thus the result set before a write would not be sorted by bucket function. With the absence of lenient ClusteredWriter this would make the test always fail for partitioned writes.